modules/fx/arrow-fx/src/main/kotlin/arrow/fx/extensions/schedule.kt

```diff
@@ -43,45 +43,39 @@ interface ScheduleApplicative<F, Input> : Applicative<SchedulePartialOf<F, Input
   fun MF(): Monad<F>
 
   override fun <A> just(a: A): Kind<SchedulePartialOf<F, Input>, A> =
-    Schedule.forever<F, A>(MF()).const(a) as Schedule<F, Input, A>
+    Schedule.once<F, A>(MF()).const(a) as Schedule<F, Input, A>
 
   override fun <A, B> Kind<SchedulePartialOf<F, Input>, A>.map(f: (A) -> B): Kind<SchedulePartialOf<F, Input>, B> =
     fix().map(f)
 }
 
 @extension
 interface ScheduleSemigroup<F, Input, Output> : Semigroup<Schedule<F, Input, Output>> {
-  fun SA(): Semigroup<Output>
+  fun OI(): Semigroup<Output>
 
   override fun Schedule<F, Input, Output>.combine(b: Schedule<F, Input, Output>): Schedule<F, Input, Output> =
-    and(b).map { (a, b) -> SA().run { a + b } }
+    and(b).map { (a, b) -> OI().run { a + b } }
 }
 
 @extension
 interface ScheduleMonoid<F, Input, Output> : Monoid<Schedule<F, Input, Output>>, ScheduleSemigroup<F, Input, Output> {
-  override fun SA(): Semigroup<Output> = MA()
-  fun MA(): Monoid<Output>
+  override fun OI(): Monoid<Output>
   fun MF(): Monad<F>
 
   override fun empty(): Schedule<F, Input, Output> =
-    Schedule.forever<F, Input>(MF()).const(MA().empty())
+    Schedule.once<F, Input>(MF()).const(OI().empty())
 }
 
 @extension
 interface ScheduleSemiring<F, Input, Output> : Semiring<Schedule<F, Input, Output>>, ScheduleMonoid<F, Input, Output> {
-  override fun MA(): Monoid<Output>
+  override fun OI(): Semiring<Output>
   override fun MF(): Monad<F>
 
-  override fun one(): Schedule<F, Input, Output> = empty()
-
-  override fun empty(): Schedule<F, Input, Output> =
-    Schedule.forever<F, Input>(MF()).const(MA().empty())
-
-  override fun zero(): Schedule<F, Input, Output> =
-    Schedule.never<F, Input>(MF()).const(MA().empty())
+  override fun one(): Schedule<F, Input, Output> =
+    Schedule.once<F, Input>(MF()).const(OI().one())
 
   override fun Schedule<F, Input, Output>.combineMultiplicate(b: Schedule<F, Input, Output>): Schedule<F, Input, Output> =
-    andThen(b).map { it.fold(::identity, ::identity) }
+    and(b).map { (a, b) -> OI().run { a * b } }
```

Comment:
The idea behind this ring instance was:
- addition is parallel composition such that every added schedule runs in parallel and is combined via `and + plus` (similar to the applicative instance which combines with `and + function application`)
- multiplication is sequential composition such that every added schedule runs after the other ones, which means combination via `andThen`

I think we can choose other instances, but since schedule has so many viable ones we should choose one which actually makes sense to use.

Here are a few monoid instances to mix and match from: ^^
- `forever().const(mempty)` + `and + plus`
- `never()` + `or + plus`
- `never()` + `andThen`
... and any other combination of the decision and the result. The decision is just a tuple of boolean and delay, so the full result that needs to be combined somehow is a triple. `and = and (on the boolean) + max (on the delay)` etc.

Reply:
the problem I had with `andThen` is that, as you said, it runs one and then the other, therefore we would need to accumulate them before multiplying the outputs.

Reply:
We should not need to multiply the outputs when using `andThen` (that'd be impossible anyway). I think this should also be lawful, but not entirely sure.